# Bonus: Temperature Analysis I

In [1]:
import pandas as pd
from datetime import datetime as dt

In [2]:
# "tobs" is "temperature observations"
df = pd.read_csv('Resources/hawaii_measurements.csv')
df.head()

station        date  prcp  tobs
0  USC00519397  2010-01-01  0.08    65
1  USC00519397  2010-01-02  0.00    63
2  USC00519397  2010-01-03  0.00    74
3  USC00519397  2010-01-04  0.00    76
4  USC00519397  2010-01-06   NaN    73

In [3]:
# Convert the date column format from string to datetime
df['date'] = pd.to_datetime(df['date'], format='%Y-%m-%d')

In [4]:
# Set the date column as the DataFrame index
df_index = df.set_index('date')


In [5]:
df_index.head()

station  prcp  tobs
date                               
2010-01-01  USC00519397  0.08    65
2010-01-02  USC00519397  0.00    63
2010-01-03  USC00519397  0.00    74
2010-01-04  USC00519397  0.00    76
2010-01-06  USC00519397   NaN    73

In [6]:
# Drop the date column



In [7]:
df.head()


station       date  prcp  tobs
0  USC00519397 2010-01-01  0.08    65
1  USC00519397 2010-01-02  0.00    63
2  USC00519397 2010-01-03  0.00    74
3  USC00519397 2010-01-04  0.00    76
4  USC00519397 2010-01-06   NaN    73

### Compare June and December data across all years 

In [8]:
from scipy import stats
import numpy as np

In [9]:
# Filter data for desired months
june = df.loc[df['date'].dt.month == 6]
december = df.loc[df['date'].dt.month == 12]



In [10]:
# Identify the average temperature for June
avg_jun = june['tobs'].mean()
avg_jun= "{:.2f}".format(avg_jun)
avg_jun

'74.94'

In [11]:
# Identify the average temperature for December
avg_dec = december['tobs'].mean()
avg_dec= "{:.2f}".format(avg_dec)
avg_dec

'71.04'

In [12]:
#check if december and june dataframes have the same shape
print(december.shape)
print(june.shape)

(1517, 4)
(1700, 4)


In [13]:
n=183
drop_june = np.random.choice(june.index, n, replace=False)
new_june = june.drop(drop_june)

In [14]:
#check if december and june dataframes have the same shape
print(december.shape)
print(new_june.shape)

(1517, 4)
(1517, 4)


In [15]:
# Create collections of temperature data
jun = new_june['tobs']
dec = december['tobs']

In [16]:
# Run paired t-test
s, p = stats.ttest_ind(jun,dec)
print(p)

5.61161960038671e-178


### Analysis

In [17]:
print(f'With a p-value of {p}, This shows that there is a difference between the two means that is statistically significant,  a p-value that is less than 0.05, is enough to be able to rejecting the null hypothesis, after compcomparing the avarages temps in hawaii, for the month of june {avg_jun}° and december {avg_dec}° it is safe to say that there is a statistically significant difference in the average temperatures in June and December in Hawaii.')

With a p-value of 5.61161960038671e-178, This shows that there is a difference between the two means that is statistically significant,  a p-value that is less than 0.05, is enough to be able to rejecting the null hypothesis, after compcomparing the avarages temps in hawaii, for the month of june 74.94° and december 71.04° it is safe to say that there is a statistically significant difference in the average temperatures in June and December in Hawaii.


The June sample had 1700 temps and December only had 1517 its best to have data samples of the sizes in order to run a paired t-test. that is why 183 temps from june were randomly selected and removed from the sample. The paired t-test was then run using the new June sample.

In [18]:
# 'worked with TA's and Other students to develop code and also looked some stuff up onon stackoverflow to help me find the answers need to get the right idea going'